napari/_qt/qt_main_window.py

```diff
@@ -447,6 +447,7 @@ def __init__(self, viewer: 'Viewer', *, show: bool = True):
         viewer.events.help.connect(self._help_changed)
         viewer.events.title.connect(self._title_changed)
         viewer.events.theme.connect(self._update_theme)
+        viewer.layers.events.connect(lambda x: self.file_menu.update())
```

Comment:
... so that here you can avoid the lambda ?

Reply:
Or (widget leak) there, maybe add that this should be disconnected on destroy of self.file_menu maybe ?